hw2/src/main/java/com/example/hw2/service/CityService.java

```diff
@@ -24,10 +27,17 @@
 	private WeatherService weatherService;
 	
 	public CityInfo getCityInfo(String cityName) {
-		
-		// TODO your code goes here
-		// delete the following line
-		return null; 
+		List<City> cities = cityRepository.findByName(cityName);
+		if (cities.size() < 1) {
+			return null;
+		}
+
+		City city = cities.get(0);
+		Country country = countryRepository.findByCode(city.getCountryCode());
+		TempAndTime tempAndTime = weatherService.getTempAndTime(cityName);
+		String time = Long.toString(tempAndTime.time);

```

Comment:
I would probably not have the time variable accessible publicly and instead use a method to access it and make time private. It was actually an issue I didn't see I had until I read your code.
```suggestion
		String time = Long.toString(tempAndTime.getTime());

```

Reply:
Implemented, thanks